From 97458f017b7f2fa3307560dee3d7fc9b7b4c96e1 Mon Sep 17 00:00:00 2001
From: YANXIAOXIH <yanxiaoxi@live.com>
Date: Wed, 17 Sep 2025 00:35:54 +0800
Subject: [PATCH] dts: rk3588-som3588-cat: Overhaul and optimize the device
 tree

This patch refactors the entire device tree for the SOM3588-Cat board.

Key improvements include:
- Cleaned up and normalized code style.
- Added descriptive comments for all major hardware nodes.
- Optimized kernel boot arguments for better performance.
- Enabled dynamic voltage scaling for PMICs to reduce power consumption.
- Fixed and enabled interrupts for the PCA9535 I/O expanders.
- Fully configured the USB Type-C port for OTG, DP Alt Mode, and PD.
- Enabled all multimedia hardware blocks (VOP, RGA, VPU, etc.).

Signed-off-by: YANXIAOXIH <yanxiaoxi@live.com>
---
 .../boot/dts/rockchip/rk3588-som3588-cat.dts  | 547 ++++++++++--------
 1 file changed, 312 insertions(+), 235 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-som3588-cat.dts b/arch/arm64/boot/dts/rockchip/rk3588-som3588-cat.dts
index d2256c50b..a564ae0eb 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-som3588-cat.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-som3588-cat.dts
@@ -1,35 +1,28 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 /*
- * Copyright (c) 2021 Rockchip Electronics Co., Ltd.
- * Copyright (c) 2023 EmbedFire <embedfire@embedfire.com>
+ * Copyright (c) 2024 Rockchip Electronics Co., Ltd.
+ *
  */
 
 /dts-v1/;
 
 #include <dt-bindings/usb/pd.h>
 #include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/pwm/pwm.h>
-#include <dt-bindings/pinctrl/rockchip.h>
-//#include <dt-bindings/input/rk-input.h>
 #include <dt-bindings/input/linux-event-codes.h>
-#include <dt-bindings/display/drm_mipi_dsi.h>
 #include <dt-bindings/display/rockchip_vop.h>
-#include <dt-bindings/sensor-dev.h>
-#include <dt-bindings/phy/jlsemi-dt-phy.h>
 #include "rk3588.dtsi"
 #include "rk3588-linux.dtsi"
 
-
 / {
 	model = "SOM3588Cat";
-	compatible = "rockchip,som588-cat", "rockchip,rk3588";
+	compatible = "rockchip,som3588-cat", "rockchip,rk3588";
 
-	/* If hdmirx node is disabled, delete the reserved-memory node here. */
+	/* 256MB CMA memory for multimedia */
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
 		ranges;
-		/* Reserve 256MB memory for hdmirx-controller@fdee0000 */
+
 		cma {
 			compatible = "shared-dma-pool";
 			reusable;
@@ -38,10 +31,12 @@ cma {
 		};
 	};
 
+	/* Kernel boot arguments */
 	chosen: chosen {
 		bootargs = "earlycon=uart8250,mmio32,0xfeb50000 console=ttyFIQ0 irqchip.gicv3_pseudo_nmi=0 rcupdate.rcu_expedited=1 rcu_nocbs=all";
 	};
 
+	/* ADC buttons */
 	adc_keys: adc-keys {
 		compatible = "adc-keys";
 		io-channels = <&saradc 1>;
@@ -56,6 +51,7 @@ power {
 		};
 	};
 
+	/* On-board LEDs */
 	leds: leds {
 		status = "okay";
 		compatible = "gpio-leds";
@@ -84,6 +80,7 @@ def_led: def-led {
 		};
 	};
 
+	/* 12V DC input */
 	vcc12v_dcin: vcc12v-dcin {
 		compatible = "regulator-fixed";
 		regulator-always-on;
@@ -93,6 +90,7 @@ vcc12v_dcin: vcc12v-dcin {
 		regulator-name = "vcc12v_dcin";
 	};
 
+	/* 5V system supply */
 	vcc5v0_sys: vcc5v0-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc5v0_sys";
@@ -103,6 +101,7 @@ vcc5v0_sys: vcc5v0-sys {
 		vin-supply = <&vcc12v_dcin>;
 	};
 
+	/* 3.3V system supply */
 	vcc3v3_sys: vcc3v3-sys {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc3v3_sys";
@@ -113,6 +112,7 @@ vcc3v3_sys: vcc3v3-sys {
 		vin-supply = <&vcc12v_dcin>;
 	};
 
+	/* 5V peripheral supply */
 	vcc_5v0: vcc-5v0 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_5v0";
@@ -123,6 +123,7 @@ vcc_5v0: vcc-5v0 {
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	/* 5V USB supply */
 	vcc5v0_usb: vcc5v0-usb {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_5v0";
@@ -133,6 +134,7 @@ vcc5v0_usb: vcc5v0-usb {
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	/* 1.8V Combo PHY analog supply */
 	combophy_avdd1v8: combophy-avdd1v8 {
 		compatible = "regulator-fixed";
 		regulator-name = "combophy_avdd1v8";
@@ -142,6 +144,7 @@ combophy_avdd1v8: combophy-avdd1v8 {
 		vin-supply = <&avcc_1v8_s0>;
 	};
 
+	/* 1.1V NLDO supply, on in suspend */
 	vcc_1v1_nldo_s3: vcc-1v1-nldo-s3 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_1v1_nldo_s3";
@@ -152,6 +155,7 @@ vcc_1v1_nldo_s3: vcc-1v1-nldo-s3 {
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	/* 3.3V supply, on when active */
 	vcc_3v3_s0: vcc-3v3-s0 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_3v3_s0";
@@ -162,6 +166,7 @@ vcc_3v3_s0: vcc-3v3-s0 {
 		vin-supply = <&vcc_3v3_s3>;
 	};
 
+	/* 3.3V SD card supply, on in suspend */
 	vcc_3v3_sd_s3: vcc-3v3-sd-s3 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_3v3_sd_s3";
@@ -172,6 +177,7 @@ vcc_3v3_sd_s3: vcc-3v3-sd-s3 {
 		vin-supply = <&vcc_3v3_s3>;
 	};
 
+	/* USB Type-C VBUS switch */
 	vbus_typec: vbus-typec {
 		compatible = "regulator-fixed";
 		regulator-name = "vbus_typec";
@@ -184,7 +190,7 @@ vbus_typec: vbus-typec {
 		pinctrl-0 = <&typec_vbus_en>;
 	};
 
-
+	/* MIPI DSI0 power */
 	mipi_dsi0_power: mipi-dsi0-power-regulator {
 		compatible = "regulator-fixed";
 		regulator-name = "mipi_dsi0_power";
@@ -194,6 +200,7 @@ mipi_dsi0_power: mipi-dsi0-power-regulator {
 		regulator-always-on;
 	};
 
+	/* MIPI DSI1 power */
 	mipi_dsi1_power: mipi-dsi1-power-regulator {
 		compatible = "regulator-fixed";
 		regulator-name = "mipi_dsi1_power";
@@ -203,6 +210,7 @@ mipi_dsi1_power: mipi-dsi1-power-regulator {
 		regulator-always-on;
 	};
 
+	/* MIPI DP1/eDP power */
 	mipi_dp1_power: mipi-edp-power-regulator {
 		compatible = "regulator-fixed";
 		regulator-name = "mipi_dp1_power";
@@ -212,6 +220,18 @@ mipi_dp1_power: mipi-edp-power-regulator {
 		regulator-always-on;
 	};
 
+	/* HDMI1 audio output */
+	hdmi1_sound: hdmi1-sound {
+		status = "okay";
+		compatible = "rockchip,hdmi";
+		rockchip,mclk-fs = <128>;
+		rockchip,card-name = "rockchip-hdmi1";
+		rockchip,cpu = <&i2s6_8ch>;
+		rockchip,codec = <&hdmi1>;
+		rockchip,jack-det;
+	};
+
+	/* HDMI audio input */
 	hdmiin_sound: hdmiin-sound {
 		compatible = "rockchip,hdmi";
 		rockchip,mclk-fs = <128>;
@@ -224,16 +244,7 @@ hdmiin_sound: hdmiin-sound {
 		rockchip,jack-det;
 	};
 
-	hdmi1_sound: hdmi1-sound {
-		status = "okay";
-		compatible = "rockchip,hdmi";
-		rockchip,mclk-fs = <128>;
-		rockchip,card-name = "rockchip-hdmi1";
-		rockchip,cpu = <&i2s6_8ch>;
-		rockchip,codec = <&hdmi1>;
-		rockchip,jack-det;
-	};
-
+	/* DisplayPort 0 audio output */
 	dp0_sound: dp0-sound {
 		status = "okay";
 		compatible = "rockchip,hdmi";
@@ -244,6 +255,7 @@ dp0_sound: dp0-sound {
 		rockchip,jack-det;
 	};
 
+	/* ES8388 audio card */
 	es8388_sound: es8388-sound {
 		status = "okay";
 		compatible = "rockchip,multicodecs-card";
@@ -288,68 +300,70 @@ next-song-key {
 		};
 	};
 
+	/* PWM cooling fan */
 	fan: pwm-fan {
 		compatible = "pwm-fan";
 		#cooling-cells = <2>;
 		pwms = <&pwm4 0 5000 0>;
 		cooling-levels = <0 100 150 200 255>;
 		rockchip,temp-trips = <
-			45000	1
-			50000	2
-			55000	3
-			60000	4
+			50000	1
+			55000	2
+			60000	3
+			65000	4
 		>;
 	};
 
+	/* Test power node */
 	test-power {
 		status = "okay";
 	};
 };
 
-// &pwm11 {
-// 	pinctrl-0 = <&pwm11m1_pins>;
-// 	status = "okay";
-// };
-
-&dmc_opp_table {
-	/delete-node/ opp-2750000000;
-};
-
+/* Enable AV1 decoder */
 &av1d {
 	status = "okay";
 };
 
+/* Enable AV1 decoder MMU */
 &av1d_mmu {
 	status = "okay";
 };
 
+/* Enable SPDIF TX2 */
 &spdif_tx2 {
 	status = "okay";
 };
 
+/* Enable I2S5 */
 &i2s5_8ch {
 	status = "okay";
 };
 
+/* Enable I2S6 */
 &i2s6_8ch {
 	status = "okay";
 };
 
+/* CPU little core supplies */
 &cpu_l0 {
 	cpu-supply = <&vdd_cpu_lit_s0>;
 	mem-supply = <&vdd_cpu_lit_mem_s0>;
 };
 
+/* CPU big core cluster 0 supplies */
 &cpu_b0 {
 	cpu-supply = <&vdd_cpu_big0_s0>;
 	mem-supply = <&vdd_cpu_big0_mem_s0>;
 };
 
+/* CPU big core cluster 1 supplies */
 &cpu_b2 {
 	cpu-supply = <&vdd_cpu_big1_s0>;
 	mem-supply = <&vdd_cpu_big1_mem_s0>;
 };
 
+/* Enable GPU */
 &gpu {
 	mali-supply = <&vdd_gpu_s0>;
 	mem-supply = <&vdd_gpu_mem_s0>;
@@ -358,28 +372,33 @@ &gpu {
 	status = "okay";
 };
 
+/* Enable DFI */
 &dfi {
 	status = "okay";
 };
 
+/* Enable DMC */
 &dmc {
 	status = "okay";
 	center-supply = <&vdd_ddr_s0>;
 	mem-supply = <&vdd_log_s0>;
 };
 
+/* Enable I2C4 */
 &i2c4 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4m3_xfer>;
 };
 
+/* Enable I2C3 */
 &i2c3 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c3m0_xfer>;
 };
 
+/* System suspend config */
 &rockchip_suspend {
 	status = "okay";
 	rockchip,sleep-debug-en = <1>;
@@ -399,6 +418,7 @@ &rockchip_suspend {
 	>;
 };
 
+/* I2C5 Bus & GPIO Expander */
 &i2c5 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c5m2_xfer>;
@@ -410,41 +430,44 @@ gpio6: gpio-expander@21 {
 		reg = <0x21>;
 		gpio-controller;
 		#gpio-cells = <2>;
-		// pinctrl-names = "default";
-		// pinctrl-0 = <&pinctrl_pca9535_1>;
-		// interrupt-parent = <&gpio2>;
-		// interrupts = <RK_PC4 IRQ_TYPE_LEVEL_HIGH>;
-		// interrupt-controller;
-		// #interrupt-cells = <2>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pca9535_1>;
+		interrupt-parent = <&gpio2>;
+		interrupts = <RK_PC4 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
 	};
 };
 
+/* I2C6 Bus & GPIO Expander */
 &i2c6 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c6m0_xfer>;
 
+	/* PCA9535 GPIO expander */
 	gpio7: gpio-expander@21 {
 		compatible = "nxp,pca9535";
 		status = "okay";
 		reg = <0x21>;
 		gpio-controller;
 		#gpio-cells = <2>;
-		// pinctrl-names = "default";
-		// pinctrl-0 = <&pinctrl_pca9535_2>;
-		// interrupt-parent = <&gpio2>;
-		// interrupts = <RK_PC5 IRQ_TYPE_LEVEL_HIGH>;
-		// interrupt-controller;
-		// #interrupt-cells = <2>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pca9535_2>;
+		interrupt-parent = <&gpio2>;
+		interrupts = <RK_PC5 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
 	};
 };
 
-
+/* I2C0 Bus & Devices */
 &i2c0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c0m2_xfer>;
 	status = "okay";
 
+	/* HYM8563 RTC */
 	hym8563: hym8563@51 {
 		compatible = "haoyu,hym8563";
 		reg = <0x51>;
@@ -458,6 +481,7 @@ hym8563: hym8563@51 {
 		wakeup-source;
 	};
 
+	/* PMIC for big core 0 */
 	vdd_cpu_big0_s0: vdd_cpu_big0_mem_s0: rk8602@42 {
 		compatible = "rockchip,rk8602";
 		reg = <0x42>;
@@ -475,6 +499,7 @@ regulator-state-mem {
 		};
 	};
 
+	/* PMIC for big core 1 */
 	vdd_cpu_big1_s0: vdd_cpu_big1_mem_s0: rk8603@43 {
 		compatible = "rockchip,rk8603";
 		reg = <0x43>;
@@ -493,11 +518,13 @@ regulator-state-mem {
 	};
 };
 
+/* I2C1 Bus & Devices */
 &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1m2_xfer>;
 	status = "okay";
 
+	/* PMIC for NPU */
 	vdd_npu_s0: vdd_npu_mem_s0: rk8602@42 {
 		compatible = "rockchip,rk8602";
 		reg = <0x42>;
@@ -515,7 +542,7 @@ regulator-state-mem {
 		};
 	};
 
-	/* Typec Controller Fusb302 */
+	/* FUSB302 Type-C Controller */
 	usbc0: fusb302@22 {
 		compatible = "fcs,fusb302";
 		reg = <0x22>;
@@ -538,6 +565,7 @@ usbc0_role_sw: endpoint@0 {
 			};
 		};
 
+		/* USB-C connector properties */
 		usb_con: connector {
 			compatible = "usb-c-connector";
 			label = "USB-C";
@@ -583,6 +611,7 @@ dp_altmode_mux: endpoint {
 	};
 };
 
+/* SPI2 Bus & Main PMIC */
 &spi2 {
 	status = "okay";
 	assigned-clocks = <&cru CLK_SPI2>;
@@ -591,6 +620,7 @@ &spi2 {
 	pinctrl-0 = <&spi2m2_cs0 &spi2m2_pins>;
 	num-cs = <1>;
 
+	/* RK806 PMIC */
 	rk806single@0 {
 		compatible = "rockchip,rk806";
 		spi-max-frequency = <1000000>;
@@ -603,21 +633,10 @@ rk806single@0 {
 		pinctrl-0 = <&pmic_pins>, <&rk806_dvs1_null>, <&rk806_dvs2_null>, <&rk806_dvs3_null>;
 		pinctrl-1 = <&rk806_dvs1_pwrdn>;
 
-		/* 2800mv-3500mv */
 		low_voltage_threshold = <3000>;
-		/* 2700mv-3400mv */
 		shutdown_voltage_threshold = <2700>;
-		/* 140 160 */
 		shutdown_temperture_threshold = <160>;
 		hotdie_temperture_threshold = <115>;
-
-		/* 0: restart PMU;
-		 * 1: reset all the power off reset registers,
-		 *	forcing the state to switch to ACTIVE mode;
-		 * 2: Reset all the power off reset registers,
-		 *	forcing the state to switch to ACTIVE mode,
-		 *	and simultaneously pull down the RESETB PIN for 5mS before releasing
-		 */
 		pmic-reset-func = <1>;
 
 		vcc1-supply = <&vcc5v0_sys>;
@@ -640,6 +659,7 @@ pwrkey {
 			status = "okay";
 		};
 
+		/* PMIC internal pin control */
 		pinctrl_rk806: pinctrl_rk806 {
 			gpio-controller;
 			#gpio-cells = <2>;
@@ -725,7 +745,10 @@ rk806_dvs3_gpio: rk806_dvs3_gpio {
 			};
 		};
 
+		/* PMIC Regulators */
 		regulators {
+
+			/* GPU core supply */
 			vdd_gpu_s0: vdd_gpu_mem_s0: DCDC_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -733,11 +756,11 @@ vdd_gpu_s0: vdd_gpu_mem_s0: DCDC_REG1 {
 				regulator-max-microvolt = <950000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_gpu_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-enable-ramp-delay = <400>;
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* CPU little core supply */
 			vdd_cpu_lit_s0: vdd_cpu_lit_mem_s0: DCDC_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -745,23 +768,21 @@ vdd_cpu_lit_s0: vdd_cpu_lit_mem_s0: DCDC_REG2 {
 				regulator-max-microvolt = <950000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_cpu_lit_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* SoC logic supply */
 			vdd_log_s0: DCDC_REG3 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <750000>;
-				regulator-max-microvolt = <750000>;
+				regulator-min-microvolt = <675000>;
+				regulator-max-microvolt = <800000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_log_s0";
-				regulator-state-mem {
-					regulator-on-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* Video encoder supply */
 			vdd_vdenc_s0: vdd_vdenc_mem_s0: DCDC_REG4 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -770,33 +791,29 @@ vdd_vdenc_s0: vdd_vdenc_mem_s0: DCDC_REG4 {
 				regulator-init-microvolt = <750000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_vdenc_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* DDR memory core supply */
 			vdd_ddr_s0: DCDC_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <850000>;
-				regulator-max-microvolt = <850000>;
+				regulator-min-microvolt = <675000>;
+				regulator-max-microvolt = <900000>;
 				regulator-ramp-delay = <12500>;
 				regulator-name = "vdd_ddr_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-					regulator-suspend-microvolt = <850000>;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* DDR VDD2 supply, on in suspend */
 			vdd2_ddr_s3: DCDC_REG6 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-name = "vdd2_ddr_s3";
-				regulator-state-mem {
-					regulator-on-in-suspend;
-				};
+				regulator-state-mem { regulator-on-in-suspend; };
 			};
 
+			/* 2.0V PLDO supply, on in suspend */
 			vcc_2v0_pldo_s3: DCDC_REG7 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -809,26 +826,25 @@ regulator-state-mem {
 				};
 			};
 
+			/* 3.3V supply, on in suspend */
 			vcc_3v3_s3: DCDC_REG8 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc_3v3_s3";
-				regulator-state-mem {
-					regulator-on-in-suspend;
-				};
+				regulator-state-mem { regulator-on-in-suspend; };
 			};
 
+			/* DDR VDDQ supply */
 			vddq_ddr_s0: DCDC_REG9 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-name = "vddq_ddr_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* 1.8V supply, on in suspend */
 			vcc_1v8_s3: DCDC_REG10 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -841,62 +857,57 @@ regulator-state-mem {
 				};
 			};
 
-			vcc_1v8_s0: PLDO_REG1 {
+			/* 1.8V analog supply */
+			avcc_1v8_s0: PLDO_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc_1v8_s0";
-				regulator-state-mem {
-					regulator-on-in-suspend;
-				};
+				regulator-name = "avcc_1v8_s0";
+				regulator-state-mem { regulator-on-in-suspend; };
 			};
 
-			avcc_1v8_s0: PLDO_REG2 {
+			/* 1.8V digital supply */
+			vcc_1v8_s0: PLDO_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				regulator-name = "avcc_1v8_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-					regulator-suspend-microvolt = <1800000>;
-				};
+				regulator-name = "vcc_1v8_s0";
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* 1.2V analog supply */
 			avdd_1v2_s0: PLDO_REG3 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1200000>;
 				regulator-max-microvolt = <1200000>;
 				regulator-name = "avdd_1v2_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* 3.3V analog supply */
 			avcc_3v3_s0: PLDO_REG4 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "avcc_3v3_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* SD card IO supply */
 			vccio_sd_s0: PLDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vccio_sd_s0";
-				regulator-state-mem {
-					regulator-off-in-suspend;
-				};
+				regulator-state-mem { regulator-off-in-suspend; };
 			};
 
+			/* PLDO6 supply, on in suspend */
 			pldo6_s3: PLDO_REG6 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -909,6 +920,7 @@ regulator-state-mem {
 				};
 			};
 
+			/* 0.75V supply, on in suspend */
 			vdd_0v75_s3: NLDO_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
@@ -921,24 +933,47 @@ regulator-state-mem {
 				};
 			};
 
+			/* DDR PLL analog supply */
 			avdd_ddr_pll_s0: NLDO_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <850000>;
 				regulator-max-microvolt = <850000>;
 				regulator-name = "avdd_ddr_pll_s0";
+				regulator-state-mem { regulator-off-in-suspend; };
+			};
+
+			/* 0.75V analog supply (at 0.8375V) */
+			avdd_0v75_s0: NLDO_REG3 {
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-min-microvolt = <837500>;
+				regulator-max-microvolt = <837500>;
+				regulator-name = "avdd_0v75_s0";
 				regulator-state-mem {
 					regulator-off-in-suspend;
-					regulator-suspend-microvolt = <850000>;
 				};
 			};
 
-			avdd_0v75_s0: NLDO_REG3 {
+			/* 0.85V analog supply */
+			avdd_0v85_s0: NLDO_REG4 {
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-min-microvolt = <850000>;
+				regulator-max-microvolt = <850000>;
+				regulator-name = "avdd_0v85_s0";
+				regulator-state-mem {
+					regulator-off-in-suspend;
+				};
+			};
+
+			/* 0.75V digital supply */
+			vdd_0v75_s0: NLDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <750000>;
 				regulator-max-microvolt = <750000>;
-				regulator-name = "avdd_0v75_s0";
+				regulator-name = "vdd_0v75_s0";
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
@@ -947,11 +982,13 @@ regulator-state-mem {
 	};
 };
 
+/* I2C7 Bus & Audio Codec */
 &i2c7 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c7m0_xfer>;
 	status = "okay";
 
+	/* ES8388 audio codec */
 	es8388: es8388@11 {
 		status = "okay";
 		#sound-dai-cells = <0>;
@@ -966,24 +1003,28 @@ es8388: es8388@11 {
 	};
 };
 
+/* Enable I2S0 */
 &i2s0_8ch {
 	status = "okay";
-	rockchip,clk-trcm = <1>;
+	rockchip,trcm-sync-tx-only;
 	pinctrl-0 = <&i2s0_lrck
 			 &i2s0_sclk
 			 &i2s0_sdi0
 			 &i2s0_sdo0>;
 };
 
+/* Enable SARADC */
 &saradc {
 	status = "okay";
 	vref-supply = <&avcc_1v8_s0>;
 };
 
+/* Enable TSADC */
 &tsadc {
 	status = "okay";
 };
 
+/* eMMC interface */
 &sdhci {
 	bus-width = <8>;
 	no-sdio;
@@ -996,6 +1037,7 @@ &sdhci {
 	status = "okay";
 };
 
+/* SD/TF card interface */
 &sdmmc {
 	max-frequency = <200000000>;
 	no-sdio;
@@ -1012,26 +1054,32 @@ &sdmmc {
 	status = "okay";
 };
 
+/* Enable IEP */
 &iep {
 	status = "okay";
 };
 
+/* Enable IEP MMU */
 &iep_mmu {
 	status = "okay";
 };
 
+/* Enable JPEG decoder */
 &jpegd {
 	status = "okay";
 };
 
+/* Enable JPEG decoder MMU */
 &jpegd_mmu {
 	status = "okay";
 };
 
+/* Enable JPEG encoder CCU */
 &jpege_ccu {
 	status = "okay";
 };
 
+/* Enable JPEG encoder 0 */
 &jpege0 {
 	status = "okay";
 };
@@ -1040,6 +1088,7 @@ &jpege0_mmu {
 	status = "okay";
 };
 
+/* Enable JPEG encoder 1 */
 &jpege1 {
 	status = "okay";
 };
@@ -1048,6 +1097,7 @@ &jpege1_mmu {
 	status = "okay";
 };
 
+/* Enable JPEG encoder 2 */
 &jpege2 {
 	status = "okay";
 };
@@ -1056,6 +1106,7 @@ &jpege2_mmu {
 	status = "okay";
 };
 
+/* Enable JPEG encoder 3 */
 &jpege3 {
 	status = "okay";
 };
@@ -1064,25 +1115,30 @@ &jpege3_mmu {
 	status = "okay";
 };
 
+/* Enable MPP service */
 &mpp_srv {
 	status = "okay";
 };
 
+/* Enable PWM4 */
 &pwm4 {
 	pinctrl-0 = <&pwm4m0_pins>;
 	status = "okay";
 };
 
+/* Enable PWM2 */
 &pwm2 {
 	pinctrl-0 = <&pwm2m0_pins>;
 	status = "okay";
 };
 
+/* Enable PWM6 */
 &pwm6 {
 	pinctrl-0 = <&pwm6m0_pins>;
 	status = "okay";
 };
 
+/* PWM3 as IR receiver */
 &pwm3 {
 	compatible = "rockchip,remotectl-pwm";
 	pinctrl-names = "default";
@@ -1092,32 +1148,33 @@ &pwm3 {
 	remote_support_psci = <0>;
 	status = "okay";
 
-	ir_key_lubancat{
+	ir_key_lemonpi{
 	rockchip,usercode = <0xff00>;
 	rockchip,key_table =
-        <0xba   KEY_POWER>,   //电源按键
-        <0xb8   KEY_MENU>,     //菜单按键
-        <0xbc   KEY_BACK>,      //返回按键
-        <0xbb   KEY_HOME>,      //home键按键
-        <0xea   KEY_PLAY>,      //播放按键
-        <0xbf   KEY_VOLUMEUP>,  //音量加按键
-        <0xe6   KEY_VOLUMEDOWN>,   //音量键按键
-        <0xf6   KEY_FASTFORWARD>,  //快进按键
-        <0xf8   KEY_FASTREVERSE>,  //快退按键
-        <0xf2   KEY_BACKSPACE>,    //BaskSpace按键
-        <0xf3   KEY_1>,            //按键1
-        <0xe7   KEY_2>,
-        <0xa1   KEY_3>,
-        <0xf7   KEY_4>,
-        <0xe3   KEY_5>,
-        <0xa5   KEY_6>,
-        <0xbd   KEY_7>,
-        <0xad   KEY_8>,
-        <0xb5   KEY_9>,
-        <0xe9   KEY_0>;            //按键0
+		<0xba   KEY_POWER>,      /* 电源按键 */
+		<0xb8   KEY_MENU>,       /* 菜单按键 */
+		<0xbc   KEY_BACK>,       /* 返回按键 */
+		<0xbb   KEY_HOME>,       /* home键按键 */
+		<0xea   KEY_PLAY>,       /* 播放按键 */
+		<0xbf   KEY_VOLUMEUP>,   /* 音量加按键 */
+		<0xe6   KEY_VOLUMEDOWN>, /* 音量键按键 */
+		<0xf6   KEY_FASTFORWARD>,/* 快进按键 */
+		<0xf8   KEY_FASTREVERSE>,/* 快退按键 */
+		<0xf2   KEY_BACKSPACE>,  /* BaskSpace按键 */
+		<0xf3   KEY_1>,          /* 按键1 */
+		<0xe7   KEY_2>,
+		<0xa1   KEY_3>,
+		<0xf7   KEY_4>,
+		<0xe3   KEY_5>,
+		<0xa5   KEY_6>,
+		<0xbd   KEY_7>,
+		<0xad   KEY_8>,
+		<0xb5   KEY_9>,
+		<0xe9   KEY_0>;          /* 按键0 */
 	};
 };
 
+/* Enable RGA3 core 0 */
 &rga3_core0 {
 	status = "okay";
 };
@@ -1126,6 +1183,7 @@ &rga3_0_mmu {
 	status = "okay";
 };
 
+/* Enable RGA3 core 1 */
 &rga3_core1 {
 	status = "okay";
 };
@@ -1134,24 +1192,29 @@ &rga3_1_mmu {
 	status = "okay";
 };
 
+/* Enable RGA2 */
 &rga2 {
 	status = "okay";
 };
 
+/* Enable NPU */
 &rknpu {
 	rknpu-supply = <&vdd_npu_s0>;
 	mem-supply = <&vdd_npu_mem_s0>;
 	status = "okay";
 };
 
+/* Enable NPU MMU */
 &rknpu_mmu {
 	status = "okay";
 };
 
+/* Enable video decoder CCU */
 &rkvdec_ccu {
 	status = "okay";
 };
 
+/* Enable video decoder 0 */
 &rkvdec0 {
 	status = "okay";
 };
@@ -1160,6 +1223,7 @@ &rkvdec0_mmu {
 	status = "okay";
 };
 
+/* Enable video decoder 1 */
 &rkvdec1 {
 	status = "okay";
 };
@@ -1168,10 +1232,12 @@ &rkvdec1_mmu {
 	status = "okay";
 };
 
+/* Enable video encoder CCU */
 &rkvenc_ccu {
 	status = "okay";
 };
 
+/* Enable video encoder 0 */
 &rkvenc0 {
 	status = "okay";
 };
@@ -1180,6 +1246,7 @@ &rkvenc0_mmu {
 	status = "okay";
 };
 
+/* Enable video encoder 1 */
 &rkvenc1 {
 	status = "okay";
 };
@@ -1188,134 +1255,115 @@ &rkvenc1_mmu {
 	status = "okay";
 };
 
+/* Enable video tunnel */
 &rkvtunnel {
 	status = "okay";
 };
 
-/* this is a watchdog */
+/* Enable watchdog */
 &wdt {
 	status = "okay";
 };
 
+/* Ethernet MAC 0 */
 &gmac0 {
-	/* Use rgmii-rxid mode to disable rx delay inside Soc */
 	phy-mode = "rgmii-rxid";
 	clock_in_out = "output";
-
 	snps,reset-gpio = <&gpio4 RK_PC6 GPIO_ACTIVE_LOW>;
 	snps,reset-active-low;
-	/* Reset time is 20ms, 100ms for rtl8211f */
 	snps,reset-delays-us = <0 20000 100000>;
-
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac0_miim
 			 &gmac0_tx_bus2
 			 &gmac0_rx_bus2
 			 &gmac0_rgmii_clk
 			 &gmac0_rgmii_bus>;
-
 	tx_delay = <0x20>;
-	/* rx_delay = <0x4f>; */
-
 	phy-handle = <&rgmii_phy0>;
 	status = "okay";
 };
 
+/* Ethernet MAC 1 */
 &gmac1 {
-	/* Use rgmii-rxid mode to disable rx delay inside Soc */
 	phy-mode = "rgmii-rxid";
 	clock_in_out = "output";
-
 	snps,reset-gpio = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>;
 	snps,reset-active-low;
-	/* Reset time is 20ms, 100ms for rtl8211f */
 	snps,reset-delays-us = <0 20000 100000>;
-
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac1_miim
 			 &gmac1_tx_bus2
 			 &gmac1_rx_bus2
 			 &gmac1_rgmii_clk
 			 &gmac1_rgmii_bus>;
-
 	tx_delay = <0x20>;
-	/* rx_delay = <0x4f>; */
-
 	phy-handle = <&rgmii_phy1>;
 	status = "okay";
 };
 
+/* MDIO0 Bus & PHY */
 &mdio0 {
 	rgmii_phy0: phy@0 {
 		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <0x00>;
-		jl2xxx,led-enable = <(JL2XXX_LED_STATIC_OP_EN | JL1XXX_LED_MODE_EN)>;
-		jl2xxx,led-mode = <(JL2XXX_LED1_LINK10  | \
-				 JL2XXX_LED1_LINK100  | \
-				 JL2XXX_LED1_LINK1000 | \
-				 JL2XXX_LED2_LINK10   | \
-				 JL2XXX_LED2_LINK100  | \
-				 JL2XXX_LED2_LINK1000 | \
-				 JL2XXX_LED2_ACTIVITY )>; //JL PHY
+		realtek,led-data = <0x6d60>;
 	};
 };
 
+/* MDIO1 Bus & PHY */
 &mdio1 {
 	rgmii_phy1: phy@0 {
 		compatible = "ethernet-phy-ieee802.3-c22";
 		reg = <0x00>;
-		jl2xxx,led-enable = <(JL2XXX_LED_STATIC_OP_EN | JL1XXX_LED_MODE_EN)>;
-		jl2xxx,led-mode = <(JL2XXX_LED1_LINK10  | \
-				 JL2XXX_LED1_LINK100  | \
-				 JL2XXX_LED1_LINK1000 | \
-				 JL2XXX_LED2_LINK10   | \
-				 JL2XXX_LED2_LINK100  | \
-				 JL2XXX_LED2_LINK1000 | \
-				 JL2XXX_LED2_ACTIVITY )>; //JL PHY
+		realtek,led-data = <0x6d60>;
 	};
 };
 
-/* PCIe30 PHY Port0 & Port1 */
+/* PCIe 3.0 PHY */
 &pcie30phy {
 	rockchip,pcie30-phymode = <PHY_MODE_PCIE_AGGREGATION>;
 	status = "okay";
 };
 
-// PCIe30 Port0 & Port1: M.2 M-Key
+/* PCIe 3.0 x4 (M.2 M-Key) */
 &pcie3x4 {
 	reset-gpios = <&gpio1 RK_PB2 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc3v3_sys>;
 	status = "okay";
 };
 
-/* SATA30_HOST0/PCIe20x1_2 Combo PHY */
-&combphy0_ps {
+/* Combo PHY 1 (PCIe/SATA) */
+&combphy1_ps {
 	status = "okay";
 };
 
-// Mini-PCIe
-&sata0 {
+/* PCIe 2.0 x1 (M.2 E-Key) */
+&pcie2x1l0 {
+	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
+	disable-gpios = <&gpio0 RK_PC6 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+/* Combo PHY 0 (PCIe/SATA) */
+&combphy0_ps {
 	status = "okay";
 };
+
+/* PCIe 2.0 x1 (Mini-PCIe) */
 &pcie2x1l2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcie_disable_pin>;
 	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
 	disable-gpios = <&gpio3 RK_PC1 GPIO_ACTIVE_HIGH>;
 	status = "disabled";
 };
 
-/* SATA30_HOST1/PCIe20x1_0 Combo PHY */
-&combphy1_ps {
-	status = "okay";
-};
-
-// M.2 E-Key
-&pcie2x1l0 {
-	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
-	disable-gpios = <&gpio0 RK_PC6 GPIO_ACTIVE_HIGH>;
+/* SATA 0 */
+&sata0{
 	status = "okay";
 };
 
-/* USB3.1/DP Combo PHY1 */
+/* USB3/DP Combo PHY 1 */
 &usbdp_phy1 {
 	status = "okay";
 };
@@ -1324,6 +1372,7 @@ &usbdp_phy1_u3 {
 	status = "okay";
 };
 
+/* USB 3.1 Host */
 &usbdrd3_1 {
 	status = "okay";
 };
@@ -1333,7 +1382,7 @@ &usbdrd_dwc3_1 {
 	dr_mode = "host";
 };
 
-/* USB3.1/DP Combo PHY0 */
+/* USB3/DP Combo PHY 0 (Type-C) */
 &usbdp_phy0 {
 	orientation-switch;
 	svid = <0xff01>;
@@ -1364,16 +1413,17 @@ &usbdp_phy0_dp {
 	status = "okay";
 };
 
-/* USB3.1 OTG0 Controller  */
+/* USB 3.1 OTG */
 &usbdrd3_0 {
 	status = "okay";
 };
 
+/* USB 3.1 OTG DWC3 core */
 &usbdrd_dwc3_0 {
 	dr_mode = "otg";
 	usb-role-switch;
-
 	status = "okay";
+	
 	port {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -1384,7 +1434,7 @@ dwc3_0_role_switch: endpoint@0 {
 	};
 };
 
-/* USB2.0 PHY0 */
+/* USB 2.0 PHY 0 (OTG) */
 &u2phy0 {
 	status = "okay";
 };
@@ -1394,6 +1444,7 @@ &u2phy0_otg {
 	status = "okay";
 };
 
+/* USB 2.0 PHY 1 (OTG) */
 &u2phy1 {
 	status = "okay";
 };
@@ -1402,12 +1453,12 @@ &u2phy1_otg {
 	status = "okay";
 };
 
-/* USB30_HOST2/SATA30_HOST2/PCIe20x1_1 Combo PHY */
+/* Combo PHY 2 (USB3/SATA/PCIe) */
 &combphy2_psu {
 	status = "okay";
 };
 
-/* USB3.1 HOST2 Controller  */
+/* USB 3.1 Host 2 */
 &usbhost3_0 {
 	status = "okay";
 };
@@ -1417,7 +1468,7 @@ &usbhost_dwc3_0 {
 	status = "okay";
 };
 
-/* USB2.0 HOST0 PHY2 */
+/* USB 2.0 PHY 2 (Host) */
 &u2phy2 {
 	status = "okay";
 };
@@ -1427,16 +1478,17 @@ &u2phy2_host {
 	status = "okay";
 };
 
-/* USB2.0 HOST0 Controller */
+/* USB 2.0 Host 0 (EHCI) */
 &usb_host0_ehci {
 	status = "okay";
 };
 
+/* USB 2.0 Host 0 (OHCI) */
 &usb_host0_ohci {
 	status = "okay";
 };
 
-/* USB2.0 HOST1 PHY3 */
+/* USB 2.0 PHY 3 (Host) */
 &u2phy3 {
 	status = "okay";
 };
@@ -1446,15 +1498,17 @@ &u2phy3_host {
 	status = "okay";
 };
 
-/* USB2.0 HOST1 Controller */
+/* USB 2.0 Host 1 (EHCI) */
 &usb_host1_ehci {
 	status = "okay";
 };
 
+/* USB 2.0 Host 1 (OHCI) */
 &usb_host1_ohci {
 	status = "okay";
 };
 
+/* VDPU (Video Decode) */
 &vdpu {
 	status = "okay";
 };
@@ -1463,25 +1517,22 @@ &vdpu_mmu {
 	status = "okay";
 };
 
+/* VEPU (Video Encode) */
 &vepu {
 	status = "okay";
 };
 
-/* dp0 display */
+/* DisplayPort 0 output */
 &dp0 {
 	status = "okay";
 };
 
+/* DP0 video routing */
 &route_dp0 {
 	status = "okay";
 	connect = <&vp1_out_dp0>;
 };
 
-&route_hdmi1{
-	status = "okay";
-	connect = <&vp0_out_hdmi1>;
-};
-
 &dp0_in_vp0 {
 	status = "disabled";
 };
@@ -1494,130 +1545,156 @@ &dp0_in_vp2 {
 	status = "disabled";
 };
 
-&hdmi1_in_vp0 {
-	status = "okay";
-};
-
-&hdmi1_in_vp1 {
-	status = "disabled";
-};
-
-&hdmi1_in_vp2 {
-	status = "disabled";
-};
-
+/* VOP (Video Output Processor) */
 &vop {
 	status = "okay";
-	// disable-win-move;
+	vop-supply = <&vdd_log_s0>;
 };
 
+/* VOP MMU */
 &vop_mmu {
 	status = "okay";
 };
 
+/* HDMI receiver */
 &hdmirx_ctrler {
 	status = "okay";
 	#sound-dai-cells = <1>;
-	/* Effective level used to trigger HPD: 0-low, 1-high */
 	hpd-trigger-level = <1>;
 	hdmirx-det-gpios = <&gpio1 RK_PD5 GPIO_ACTIVE_LOW>;
 	pinctrl-0 = <&hdmim1_rx_cec &hdmim1_rx_hpdin &hdmim1_rx_scl &hdmim1_rx_sda &hdmirx_det>;
 	pinctrl-names = "default";
 };
 
-/* HDMI IN Audio */
+/* Display subsystem */
+&display_subsystem {
+	clocks = <&hdptxphy_hdmi1>;
+	clock-names = "hdmi1_phy_pll";
+};
+
+/* HDMI1 TX PHY */
+&hdptxphy_hdmi1 {
+	status = "okay";
+};
+
+/* HDMI1 video routing */
+&route_hdmi1{
+	status = "okay";
+	connect = <&vp0_out_hdmi1>;
+};
+
+/* HDMI1 output */
+&hdmi1 {
+	enable-gpios = <&gpio1 RK_PC6 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+&hdmi1_in_vp0 {
+	status = "okay";
+};
+
+&hdmi1_in_vp1 {
+	status = "disabled";
+};
+
+&hdmi1_in_vp2 {
+	status = "disabled";
+};
+
+
+/* HDMI-in audio link */
 &hdmiin_sound {
 	status = "okay";
 };
 
+/* I2S7 for HDMI-in audio */
 &i2s7_8ch{
 	status = "okay";
 };
 
-/* vp0 & vp1 splice for 8K output */
+/* Video Plane 0 (for 8K) */
 &vp0 {
 	rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER0 | 1 << ROCKCHIP_VOP2_ESMART0)>;
 	rockchip,primary-plane = <ROCKCHIP_VOP2_ESMART0>;
 	// cursor-win-id = <ROCKCHIP_VOP2_CLUSTER0>;
 };
 
+/* Video Plane 1 (for 8K) */
 &vp1 {
 	rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER1 | 1 << ROCKCHIP_VOP2_ESMART1)>;
 	rockchip,primary-plane = <ROCKCHIP_VOP2_ESMART1>;
 	cursor-win-id = <ROCKCHIP_VOP2_CLUSTER1>;
 };
 
+/* Video Plane 2 */
 &vp2 {
 	rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER2 | 1 << ROCKCHIP_VOP2_ESMART2)>;
 	rockchip,primary-plane = <ROCKCHIP_VOP2_ESMART2>;
 	cursor-win-id = <ROCKCHIP_VOP2_CLUSTER2>;
 };
 
+/* Video Plane 3 */
 &vp3 {
 	rockchip,plane-mask = <(1 << ROCKCHIP_VOP2_CLUSTER3 | 1 << ROCKCHIP_VOP2_ESMART3)>;
 	rockchip,primary-plane = <ROCKCHIP_VOP2_ESMART3>;
 	cursor-win-id = <ROCKCHIP_VOP2_CLUSTER3>;
 };
 
-&hdmi1 {
-	enable-gpios = <&gpio1 RK_PC6 GPIO_ACTIVE_HIGH>;
-	status = "okay";
-};
-
-&display_subsystem {
-	clocks = <&hdptxphy_hdmi0>, <&hdptxphy_hdmi1>;
-	clock-names = "hdmi0_phy_pll", "hdmi1_phy_pll";
-};
-
-&hdptxphy_hdmi0 {
-	status = "okay";
-};
-
-&hdptxphy_hdmi1 {
-	status = "okay";
-};
-
+/* Pin control hub */
 &pinctrl {
+	
+	/* RTC interrupt pin */
 	hym8563 {
 		hym8563_int: hym8563-int {
 			rockchip,pins = <0 RK_PB0 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
 
+	/* Headphone detect pin */
 	headphone {
 		hp_det: hp-det {
 			rockchip,pins = <1 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
 
+	/* Pins for USB Type-C */
 	usb-typec {
 		fusb302_int: fusb302-int {
 			rockchip,pins = <0 RK_PC4 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
-
 		typec_vbus_en: typec-vbus-en {
 			rockchip,pins = <2 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
 
+	/* System LED pin */
 	leds {
 		sys_led_pin: sys-led-pin {
 			rockchip,pins = <0 RK_PD3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
 
+	/* HDMI-in detect pin */
 	hdmirx {
 		hdmirx_det: hdmirx-det {
 			rockchip,pins = <1 RK_PD5 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
 
+	/* IO expander interrupt pins */
 	io {
-		pinctrl_pca9535_1: pinctrl-pca9535-1{
+		pinctrl_pca9535_1: pinctrl-pca9535-1 {
 			rockchip,pins = <2 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
-		pinctrl_pca9535_2: pinctrl-pca9535-2{
+		pinctrl_pca9535_2: pinctrl-pca9535-2 {
 			rockchip,pins = <2 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	/* PCIe disable pin */
+	pcie {
+		pcie_disable_pin: pcie-disable-pin{
+			rockchip,pins = <3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
-- 
2.39.5

